### PR TITLE
Validate etcd image versions in test manifests

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind cleanup
/sig architecture testing
/area code-organization

#### What this PR does / why we need it:

Ensures that the test images for etcd stay in sync.

#### Which issue(s) this PR fixes:

Fixes the issue not caught and hence fixed by https://github.com/kubernetes/kubernetes/pull/106562

#### Special notes for your reviewer:

See also https://kubernetes.slack.com/archives/C09QZ4DQB/p1637362487110300

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: build/dependencies.yaml [hunk 1]
```diff
@@ -75,6 +75,8 @@ dependencies:
       match: quay.io/coreos/etcd
     - path: test/e2e/framework/nodes_util.go
       match: const etcdImage
+    - path: test/utils/image/manifest.go
+      match: configs\[Etcd\] = Config{list\.GcEtcdRegistry, "etcd", "\d+\.\d+.\d+(-(alpha|beta|rc).\d+)?(-\d+)?"}
```
Comment (ehashman) [inline build/dependencies.yaml:79]: Attempted to make this match the etcd versioning scheme in https://github.com/etcd-io/etcd/tags, I'm not sure if it cares about the "-0" on the container image version but seems like no

**Comment by ehashman** [comment]
Test is correctly failing, should pass after https://github.com/kubernetes/kubernetes/pull/106562 merges

File: build/dependencies.yaml (hunk 1 above)
Comment (SergeyKanzhelev) [reply build/dependencies.yaml:79]: this version will be in the manifest: https://github.com/kubernetes/kubernetes/blob/0c6f1b4e2ac06da94e904e86ad5dcef81b0f8ed0/cluster/images/etcd/Makefile#L40

Comment (SergeyKanzhelev) [reply build/dependencies.yaml:79]: so revision is needed, it is coming from k8s, not from etcd

Comment (SergeyKanzhelev) [reply build/dependencies.yaml:79]: at least in the file I updated for e2e

Comment (BenTheElder) [reply build/dependencies.yaml:79]: FWIW this tool is just going to check for a substring match with the `version` field after identifying the line that matches this rule. so it's also OK to write a less complex match regex.

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve
/hold
(hold is to prevent /retest spam from the robot until the fix PR is in)

**Comment by BenTheElder** [comment]
IMO
/kind bug
It is a bug that we were not matching this version fully. This fix should go in.
/triage accepted
/priority important-soon.

**Comment by BenTheElder** [comment]
test fix, tagging for milestone.

File: build/dependencies.yaml [hunk 2]
```diff
@@ -75,6 +75,8 @@ dependencies:
...
 
   - name: "etcd-image"
```
Comment (BenTheElder) [inline build/dependencies.yaml:81]: leaving a breadcrumb (https://kubernetes.slack.com/archives/C09QZ4DQB/p1637362487110300), we need to move this down to the ectd-image group at some point (my bad!)

**Pull request merged**